test/e2e/azure_logcollector.go

```diff
@@ -183,10 +183,21 @@ func getAzureCluster(ctx context.Context, managementClusterClient client.Client,
 	}
 
 	azCluster := &v1beta1.AzureCluster{}
-	err := managementClusterClient.Get(context.TODO(), key, azCluster)
+	err := managementClusterClient.Get(ctx, key, azCluster)
 	return azCluster, err
 }
 
+func getAzureManagedControlPlane(ctx context.Context, managementClusterClient client.Client, namespace, name string) (*expv1alpha4.AzureManagedControlPlane, error) {
+	key := client.ObjectKey{
+		Namespace: namespace,
+		Name:      name,
+	}
+
+	azManagedControlPlane := &expv1alpha4.AzureManagedControlPlane{}
```

Comment:
There is an issue with the import name, this shouldn't be `expv1alpha4`, it's using the v1beta1 API version

Reply:
#2574 will address the import alias, but either this PR or that one will need an update here when the other is merged. Should I mark this PR as blocked on that one to make sure we don't forget?

Reply:
Either way is fine, I can easily rebase #2574 if this merges first.